src/Elasticsearch/ConnectionPool/SniffingConnectionPool.php

```diff
@@ -117,7 +118,12 @@ private function sniffConnection(Connection $connection): bool
             return false;
         }
 
-        $nodes = $this->parseClusterState($connection->getTransportSchema(), $response);
+        try {
+            $nodes = $this->parseClusterState($connection->getTransportSchema(),
```

Comment:
in the `parseClusterState` function 
```
private function parseClusterState(string $transportSchema, $nodeInfo): array
```
the `nodeInfo` parameter is a object of `GuzzleHttp\Ring\Future\FutureArray`, when we get some attribute, we will call the `__get` magic method， then call the `wait` function  in https://github.com/guzzle/RingPHP/blob/5e2a174052995663dd68e6b5ad838afd47dd615b/src/Future/BaseFutureTrait.php#L63 it will throw the error.

Reply:
@lawlielt yes, you right good catch! Looks good. Btw,sorry for my late reply.